pkgs/by-name/lu/luau-lsp/package.nix

```
@@ -0,0 +1,60 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "luau-lsp";
+  version = "1.45.0";
+
+  src = fetchFromGitHub {
+    owner = "JohnnyMorganz";
+    repo = "luau-lsp";
+    tag = finalAttrs.version;
+    hash = "sha256-OJAjTy0vTRb43TTiPeXafWq4kjIpnDXoTprVzbMnaWQ=";
+    fetchSubmodules = true;
+  };
+
+  NIX_CFLAGS_COMPILE = "-Wno-error";
+
+  cmakeFlags = lib.optionals stdenv.hostPlatform.isDarwin [
+    (lib.cmakeFeature "CMAKE_OSX_ARCHITECTURES" stdenv.hostPlatform.darwinArch)
+  ];
+
+  nativeBuildInputs = [ cmake ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    cmake --build . --target Luau.LanguageServer.CLI --config Release
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D luau-lsp $out/bin/luau-lsp
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Language Server Implementation for Luau";
+    homepage = "https://github.com/JohnnyMorganz/luau-lsp";
+    downloadPage = "https://github.com/JohnnyMorganz/luau-lsp/releases/tag/${finalAttrs.version}";
+    changelog = "https://github.com/JohnnyMorganz/luau-lsp/blob/${finalAttrs.version}/CHANGELOG.md";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [
+      anninzy
+      HeitorAugustoLN
+    ];
+    mainProgram = "luau-lsp";
+    platforms = lib.platforms.all;
```

Let's mark `x86_64-darwin` as broken for now, and merge this PR, as @teto suggested

```suggestion
    platforms = lib.platforms.all;
    broken = stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isx86_64;
```

---

That's better, what is the difference between them?